### ISSUE TITLE
ID3 supports duplicates of the same frame type why not support it?

### ISSUE THREAD

**Issue by Clementine-Issue-Importer** [clementine-player/Clementine#409]
_From [yellow_onion@orcon.net.nz](https://code.google.com/u/100939741841532727985/) on June 15, 2010 10:52:22_

the other day I learnt that ID3 supports multiple frames of the same type

explanation:
I'm going through tagging all my music, and I've come across the ability (with MusicBrains Picard and the Lastfm Plus Plugin) to add multiple genre tags to my music. This is what I've been looking for for a while, this enables us to restructure the group by genre structure of the library to cross reference songs with the genre's and list them, One, two three, or how many time's you have a the genre tag, or you could use the genre tag as a hierarchy of genre's, sub-genre's and sub-sub-genre's and so on.

the next feature would be cross referencing Artist's using the same format as the first suggestion plus a seperator tag( put in a TXXX frame) so you save the Artist as ['Artist 1','Artist 2'] and then save in the TXXX frame a 'vs'(I have a lot of these Vs songs), 'and' or ';'

Songs like "My Own Summer (Shove It)" feature on many albums like The Matrix OST and Around the Fur, could benefit from using two frames for the album and since the matrix OST is a Various artist album, the Album Artist Tag could benefit from having two frames

So your probably wondering what formats support duplicate tags
ID3v2.3(I think) and up  supports lists in most tags, I personally can't find a tag that doesn't support it( this includes FLAC and MP3, RIFF, TrueAudio)
Xiph Comments supports it( FLAC, and Vorbis)
APE doesn't seem to support it ( mp3, Wavpack)
MP4 doesn't support it (AAC, anything else you wanna pack in it I guess?)
ASF doesn't support it

this is going by reference of the taglib implementation 

I mean this will require an over haul of the database engine (I already have a SQL scheme in the works but that's for another project I'm working on) and probably deviates away for the Amarok Clone Model (personally I Liked Amarok's UI the most and the database could have used a redesign

_Original issue: http://code.google.com/p/clementine-player/issues/detail?id=409_

**Comment by Kaemmelot** [comment]
Will this change in the near future? Because this would be such an important feature for me.

**Comment by Beckfield** [comment]
I just discovered Clementine because I was experimenting with single-file flac rips with CUE sheets.  This is a really nice player, but it's out as the go-to library manager, for this very issue.  With services like Gracenote plugging multiple genres into almost everything, managing a library with a manager that doesn't support it is simply not sustainable.

**Comment by hamjo** [comment]
I would really like to have this feature. Also, I would add that Clementine shoudl also support mulitple tags for artists, since multiple artists can work on the same song. Since there is no commit associated to this issue (correct me if I'm wrong), I will try to work on it on my free time. I don't want to prevent anyone to work on it though.

**Comment by cjsthompson** [comment]
@hamjo I've been waiting a long time for a library-oriented player to support this. The Pragha dev is interested in adding this feature but he says what's blocking him is support in taglib. I'm pretty sure Clementine uses taglib too. So perhaps a taglib feature request might unblock the situation.

**Comment by hamjo** [comment]
I am sorry about what I proimised, but I won't be implementing this feature. Since my computer would be the only one to have a music player supporting multiple values for a tag, I do not have an interest anymore into that feature. Sorry.

**Comment by Sughosha** [comment]
I came here looking for the same issue, support for multiple frames. I was using [MusicBee](https://getmusicbee.com/) and [MP3tag](https://www.mp3tag.de/) for tagging purpose, until I started using Linux. In Linux I am using [puddletag](http://puddletag.sourceforge.net) to tag my songs. I have tagged all my songs with multiple frames for multiple values (artists, composers, lyricists, etc.). Both [ID3](https://id3.org/id3v2.4.0-frames) and [Vorbis Comment](https://xiph.org/vorbis/doc/v-comment.html) can contain multiple frames for same tag.

In MusicBee I had no problems as it handles multiple values like this:
![Screenshot  MusicBee](https://user-images.githubusercontent.com/59734562/84415708-cadb5180-ac30-11ea-8d90-94531dce59d2.png)
As you can see here, it just shows a ";" to represent the values of different frames.

In Clementine, only the first artist, of the same song, which is mentioned in the above screenshot, is shown, and the second artist is ignored and not listed in the Artists column. If I edit the tags, it shows as "Adam Szabo Johan Vilborg" as if there is only one artist with a big name.

What I would recommend is, to add at least a ";", like MusicBee does, to represent and differentiate all the artists, instead of showing all of them without any delimiter as if there is only a single artist with a big name.

**Comment by slowhand73** [comment]
Is this have been put as a feature request ? Has it been acknowledge by the Clementine devs ?
I would love to see this since I started a few month ago to tag my CD collection converted in FLAC with Muzicbrain's Picard.
Artists, genres and so on can be multi-valued and I think it is a great feature (and Clementine is my favorite music player).
Thanks.
